src/location/location.service.ts

```diff
@@ -0,0 +1,86 @@
+import { CacheService } from '../cache/cache.service';
+import { InjectModel } from '@nestjs/mongoose';
+import { Injectable } from '@nestjs/common';
+import { Location, LocationDocument } from './schemas/location.schema'; // Import the CacheService
+import { Model } from 'mongoose';
+
+@Injectable()
+export class LocationService {
+  constructor(
+    @InjectModel(Location.name) private _locationModel: Model<LocationDocument>,
+    private _cacheService: CacheService
+  ) {}
+
+  async findLocation(postal_code: string): Promise<Location | undefined> {
+    return this._locationModel.findOne({ PostalCode: postal_code });
+  }
+
+  async getAllLocation(): Promise<Location[]> {
+    return this._locationModel.find().exec();
+  }
+
+  async getNearbyPostalCodes(
+    distance: number,
+    postalCode: string,
+  ): Promise<Location[]> {
+    const cacheKey = `nearbyPostalCodes:${postalCode}:${distance}`;
```

Comment:
Add a comment of the unit for distance, or add it in the cacheKey

Reply:
added comment regarding units for distance.

For now, we just use km. We can implement miles in the future if needed.